ttnn/cpp/ttnn/operations/transformer/sdpa/device/sdpa_op.cpp

```diff
@@ -62,11 +66,18 @@ void ScaledDotProductAttention::validate(
         // Check qkv heads
         TT_FATAL(q_shape[-3] >= k_shape[-3]);
 
-        TT_FATAL(mask_shape[-3] == 1);
+        if (mask_option.has_value()){
+            const auto mask_shape = mask_option.value().get_legacy_shape();
+            TT_FATAL(q_shape[-4] == mask_shape[-4]);
+            TT_FATAL(q_shape[-2] == mask_shape[-2] && q_shape[-2] == mask_shape[-1]);
+            TT_FATAL(mask_shape[-3] == 1);
```

Comment:
Same as above. Python users who will see error message generated by current code will find it useless.